server/utils/address.ts

```diff
@@ -0,0 +1,29 @@
+export default class Address {
+  constructor(
+    readonly addressLine1: string,
+    readonly addressLine2: string,
+    readonly townOrCity: string,
+    readonly county: string,
+    readonly postcode: string
+  ) {}
+
+  static fromArray(address: string[] | null): Address | null {
+    if (address === null) {
+      return null
+    }
+    if (address.length !== 5) {
+      return null
+    }
+    return new Address(address[0], address[1], address[2], address[3], address[4])
```

Comment:
this slightly worries me. i do actually like that the main portion of the address is unstructured, but then unpacking an array into a structured data class like this seems error prone. perhaps i missed the discussions on this, but doesn't this tightly couple the UI implementation to the data model?

Reply:
i've just continued to read below and i see that the address form _is_ structured, so perhaps i'm just not grepping the full flow end to end. but it all looks fine where the address form is parsed.

Reply:
We had a discussion about this a while ago and we decided that we wanted to make the address fields agnostic to the type for instance, npsOfficeCode, otherLocationsAddress and videoCallUrl (which has now been removed).

This means that for the JSON itself sending to the server will be agnostic.

I have the issue whereby once I have a `requirement` defined in my mind I don't question it as heavily; I think perhaps this requirement was super-ceded with another design.

I do prefer to have explicit fields for the address rather than an array. If you are in agreement, I will change to using fields rather than array.